transformation/src/main/java/org/opensearch/migrations/transformation/rules/StringMappingType.java

```diff
@@ -0,0 +1,98 @@
+package org.opensearch.migrations.transformation.rules;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+import org.opensearch.migrations.transformation.CanApplyResult;
+import org.opensearch.migrations.transformation.TransformationRule;
+import org.opensearch.migrations.transformation.entity.Index;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Supports transformation of the mapping of 'string' -> 'text' that were changed between ES 2 to ES 5
+ *
+ * Example:
+ * Starting state (ES 2):
+ *  "mappings": {
+ *      "properties": {
+ *          "field1": {
+ *             "type": "string"
+ *          }
+ *       }
+ *  }
+ *
+ * Ending state (ES 5):
+*  "mappings": {
+ *     "properties": {
+ *        "field1": {
+ *           "type": "text",
+ *           "fields": {
+ *              "keyword": {
+ *                 "type": "keyword",
+ *                 "ignore_above": 256
```

Comment:
To yesterday's discussion on transformation paths -- are there cases where these are not the settings users will want? Are these settings the closest possible match for the `string` behavior or one of a couple plausible variations?

Reply:
Note: This value is the default value in modern OpenSearch for text based fields.

I don't think this catch all mechanism is the best approach to expose configurability. Customers that want to tweak these values would want to specific field names with different values.

I'll create a ticket this kind of scenario

Reply:
Created [MIGRATIONS-2435 | Define user experience for handling many breaking changes](https://opensearch.atlassian.net/browse/MIGRATIONS-2435)